RealmSwift/RealmConfiguration.swift

```diff
@@ -184,7 +184,19 @@ extension Realm {
 
          - note: Setting this property to `true` doesn't disable file format migrations.
          */
-        public var deleteRealmIfMigrationNeeded: Bool = false
+        public var deleteRealmIfMigrationNeeded: Bool {
+            get {
+                return _deleteRealmIfMigrationNeeded
+            }
+            set(newValue) {
+                if newValue && syncConfiguration != nil {
+                    throwRealmException("Cannot set 'deleteRealmIfMigrationNeeded' when sync is enabled ('syncConfig' is set).")
```

Comment:
A throw on a computed property isn't correct, also anything that should should declare `throws`. RLMRealmConfiguration will perform the throw.

Reply:
throwRealmException throws an obj-c exception and not a swift error, so `throws` isn't applicable.